python/tests/scripts/test_update_storage.py

```diff
@@ -37,139 +39,97 @@ def _get_azure_storage_config(cfg):
 
 
-def test_upgrade_script_dryrun_s3(moto_s3_endpoint_and_credentials):
-    # Given
-    endpoint, port, bucket, aws_access_key, aws_secret_key = moto_s3_endpoint_and_credentials
-    uri = (
-        endpoint.replace("http://", "s3://").rsplit(":", 1)[0]
-        + ":"
-        + bucket
-        + "?access="
-        + aws_access_key
-        + "&secret="
-        + aws_secret_key
-        + "&port="
-        + port
-    )
-
-    ac = Arctic(uri)
+def test_upgrade_script_dryrun_s3(s3_storage: S3Bucket):
+    ac = s3_storage.create_arctic()
     create_library_config(ac, LIB_NAME)
 
     # When
-    run(uri=uri, run=False)
+    run(uri=s3_storage.arctic_uri, run=False)
 
     # Then
     config = ac._library_manager.get_library_config(LIB_NAME)
-    s3_storage = _get_s3_storage_config(config)
-    assert s3_storage.bucket_name == bucket
-    assert s3_storage.credential_name == aws_access_key
-    assert s3_storage.credential_key == aws_secret_key
-
-
-def test_upgrade_script_s3(moto_s3_endpoint_and_credentials):
-    endpoint, port, bucket, aws_access_key, aws_secret_key = moto_s3_endpoint_and_credentials
-    uri = (
-        endpoint.replace("http://", "s3://").rsplit(":", 1)[0]
-        + ":"
-        + bucket
-        + "?access="
-        + aws_access_key
-        + "&secret="
-        + aws_secret_key
-        + "&port="
-        + port
-    )
+    storage_config = _get_s3_storage_config(config)
+    assert storage_config.bucket_name == s3_storage.bucket
+    assert storage_config.credential_name == s3_storage.key.id
+    assert storage_config.credential_key == s3_storage.key.secret
 
-    ac = Arctic(uri)
+
+def test_upgrade_script_s3(s3_storage: S3Bucket):
+    ac = s3_storage.create_arctic()
     create_library_config(ac, LIB_NAME)
 
-    run(uri=uri, run=True)
+    run(uri=s3_storage.arctic_uri, run=True)
 
     config = ac._library_manager.get_library_config(LIB_NAME)
-    s3_storage = _get_s3_storage_config(config)
-    assert s3_storage.bucket_name == ""
-    assert s3_storage.credential_name == ""
-    assert s3_storage.credential_key == ""
+    storage_config = _get_s3_storage_config(config)
+    assert storage_config.bucket_name == ""
+    assert storage_config.credential_name == ""
+    assert storage_config.credential_key == ""
 
 
-def get_s3_aws_auth_uri_bucket_and_setup_server(moto_s3_endpoint_and_credentials, monkeypatch):
-    endpoint, port, bucket, aws_access_key, aws_secret_key = moto_s3_endpoint_and_credentials
-    monkeypatch.setenv("AWS_ACCESS_KEY_ID", aws_access_key)
-    monkeypatch.setenv("AWS_SECRET_ACCESS_KEY", aws_secret_key)
-    return (
-        endpoint.replace("http://", "s3://").rsplit(":", 1)[0] + ":" + bucket + "?aws_auth=true" + "&port=" + port,
-        bucket,
-    )
+def test_upgrade_script_s3_rbac_ok(s3_storage: S3Bucket, monkeypatch):
+    """Just _RBAC_ as creds is a placeholder. Leave config with that alone."""
+    if os.name == "nt":
+        if sys.version_info < (3, 9):
```

Comment:
Awesome investigation here, thank you. This test was being skipped on Windows before.